test/main.test.js

```diff
@@ -129,4 +130,50 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('doesn\'t mix up contexts if using concurrently then() from when 3.7.7',
+  function() {
+    expect(require('when/package.json').version).to.equal('3.7.7');
```

Comment:
PTAL
Solved in a different way (local modules and stub modules)
Inspiration from here:
Local modules http://stackoverflow.com/a/26028854
Stub modules https://github.com/npm/npm/issues/5499#issuecomment-71089080
Works for me using two different, conflicting versions of the same package.